devdoc/WorkUnits.rst

```diff
@@ -430,12 +430,55 @@ Once all of the required workflow items are executed, the workunit is marked as
 workflow items that are waiting to be triggered by an event, the workunit will be passed to the scheduler,
 which will keep monitoring for events.
 
+Note that most items in the xml workflow begin in the state WFStateNull. This means that it is valid to execute them,
+but they haven't been executed yet. Typically, only a few items begin with the state WFStateReqd.
+
 There are various specialised types of workflow items - e.g. sequential, wait, independent, but they all follow the same basic approach of
 executing dependencies and then executing that particular item.
 
 Most of the interesting work in an ECL query is done within a graph.  The call ctx->executeGraph will either execute the graph locally
 (in the case of hthor and roxie), or add the workunit onto a queue (for Thor).  Whichever happens, control will pass to that engine.
 
+Specialised Workflow Items
+=================
+Each item mode/type can affect the dependency structure of the workflow:
+
+- Sequential/Ordered
+
+  The workflow structure for sequential and ordered is the same. An item is made to contain all of the actions in the statement. This is
+  achieved by making each action a dependency of this item. The dependencies, and consequently the actions, must be executed in order.
+  An extra item is always inserted before each dependency. This means that if other statements reference the same attribute, it will only
```

Comment:
attribute -> dependent workflow item.  If it isn't a workflow item it will be executed more than once.